pgmpy/datasets/datasets.py

```diff
@@ -0,0 +1,342 @@
+from __future__ import annotations
+
+import hashlib
+import io
+import os
+import re
+from pathlib import Path
+from typing import Any, Dict, Optional, Tuple
+
+import networkx as nx
+import pandas as pd
+import requests
+
+from pgmpy.base import DAG
+
+from .registry import DATASETS, dataset_class
+
+PGMPY_DATA_HOME = os.environ.get(
+    "PGMPY_DATA_HOME", os.path.join(Path.home(), ".pgmpy", "data")
+)
+
+
+def ensure_dir(path: str) -> None:
+    Path(path).mkdir(parents=True, exist_ok=True)
+
+
+def load_df_from_txt(raw: bytes) -> pd.DataFrame:
+    try:
+        return pd.read_csv(io.BytesIO(raw), sep=None, engine="python")
+    except Exception as e:
+        try:
+            return pd.read_csv(io.BytesIO(raw), sep=",", engine="python")
+        except Exception:
+            raise e
```

Comment:
Please don't use try-except blocks. Unexpected events should be handled explicitly. Should look into the reason why this could throw an error and that case should be handled explicitly.